server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStore.java

```diff
@@ -245,26 +245,31 @@ public TokenCode mapRow(ResultSet rs, int rowNum) throws SQLException {
             String userid = rs.getString(pos++);
             String client_id = rs.getString(pos++);
             long expiresat = rs.getLong(pos++);
-            Timestamp created = rs.getTimestamp(pos++);
+            Instant created = rs.getTimestamp(pos++).toInstant();
             byte[] authentication = rs.getBytes(pos++);
-            return createTokenCode(code, userid, client_id, expiresat, created, authentication);
+
+            Optional<Instant> expirationInstant =
+                    (expiresat == 0) ?
+                            Optional.empty() :
+                            Optional.of(Instant.ofEpochMilli(expiresat));
+
+            return createTokenCode(code, userid, client_id, expirationInstant, created, authentication);
         }
     }
 
-    public TokenCode createTokenCode(String code, String userId, String clientId, long expiresAt, Timestamp created, byte[] authentication) {
+    public TokenCode createTokenCode(String code, String userId, String clientId, Optional<Instant> expiresAt, Instant created, byte[] authentication) {
```

Comment:
[Call "Optional#isPresent()" or "!Optional#isEmpty()" before accessing the value.](https://sonarcloud.io/project/issues)

Reply:
I agree. I was planning an improvement and I forgot to do it. Coming up!

Thanks for catching this.

Reply:
I fully acknowledge that I didn't technically fix this, though the method is only for testing purposes. So it seems to me that it is OK in this case. Let me know what you think.